File: domain/src/main/java/com/thinkzi/weather/domain/exception/DefaultErrorBundle.java
package com.thinkzi.weather.domain.exception;

/**
* provide Wrapper around Exceptions used to manage default errors
* */
public class DefaultErrorBundle implements ErrorBundle {

    private static final String DEFAULT_ERROR_MSG = "Unknown error";

    private final Exception _exception;

    public DefaultErrorBundle(Exception _exception) {
        this._exception = _exception;
    }

    @Override
    public Exception getException() {
        return _exception;
    }

    @Override
    public String getErrorMessage() {
        return (_exception != null) ? this._exception.getMessage() : DEFAULT_ERROR_MSG;
    }

}

File: domain/src/main/java/com/thinkzi/weather/domain/exception/DefaultErrorBundle.kt
package com.thinkzi.weather.domain.exception

/**
 * provide Wrapper around Exceptions used to manage default errors
 */
class DefaultErrorBundle(private val _exception: Exception) : ErrorBundle {
    override val exception: Exception
        get() = _exception
    override val errorMessage: String
        get() = this._exception.message!!

    companion object {

        private val DEFAULT_ERROR_MSG = "Unknown error"
    }

}
